File: Pattern39.py
n = int(input("enter a number: "))
if(n>2):
    add = n-3
else:
    add=0
rows = n+add
columns = n*2-1
for i in range(0,rows):
    for j in range(0,columns):
        #the conditions 
        if((j == n-1-i) or (j == n-1+i) or (i>n-1 and j == columns-1 ) or (i>n-1 and j == 0) or (i == n-1 and j%2 == 0 )):
            print('*',end=' ')
        else:
            print(' ',end=' ')
    print()

File: Pattern39.c
#include <stdio.h>

int main()
{
    int i, j, k, rows;

    printf("Enter a number: ");
    scanf("%d", &rows);

    for (i = 0; i < rows; i++)
    {
        for (j = 0; j < rows-1-i; j++)
        {
            printf(" ");
        }

        for (k = 0; k < i+1; k++)
        {
            if (i>1 && i < rows-1 && k > 0 && k < i)
            printf("  ");
            else
            printf("* ");
        }
        printf("\n");
    }

    for (i = 0; i < rows - 3; i++)
    {
        for (j = 0; j < rows; j++)
        {
            if (j == 0 || j == rows - 1)
                printf("* ");
            else
                printf("  ");
        }
        printf("\n");

    }
    return 0;
}
